### PR TITLE
Add test for YQL common libraries.

### PR DESCRIPTION
**Pull request by Tony-Romanov**

### CODE REVIEW

**Comment by Tony-Romanov** [comment]
https://github.com/ydb-platform/ydb/issues/9398

File: yt/yql/tests/conftest.py [hunk 1]
```diff
@@ -21,14 +21,15 @@ def _get_artifacts_path():
...
-    def __init__(self, env, count):
+    def __init__(self, env, count, libraries):
         self.yql_agent = YqlAgentComponent()
 
         self.yql_agent.prepare(env, config={
             "count": count,
             "path": yatest.common.binary_path("yt/yql/agent/bin"),
             "artifacts_path": _get_artifacts_path(),
             "native_client_supported": True,
+            "libraries": libraries
```
Comment (Kontakter) [inline yt/yql/tests/conftest.py:32]: Let add comma in the end of line

File: yt/python/yt/environment/components/yql_agent/__init__.py [hunk 2]
```diff
@@ -27,6 +27,8 @@ def prepare(self, env, config):
         if "path" in config:
             self.PATH = config["path"]
 
+        self.libraries = config["libraries"] if "libraries" in config else {}
```
Comment (Kontakter) [inline yt/python/yt/environment/components/yql_agent/__init__.py:30]: Let write shorter: `config.get("libraries", {})`

File: yt/yql/tests/conftest.py [hunk 3]
```diff
@@ -56,6 +57,15 @@ def update_yql_agent_environment(cls, yql_agent):
...
     count = getattr(cls, "NUM_YQL_AGENTS", 1)
-    with YqlAgent(cls.Env, count) as yql_agent:
+
+    libraries = {}
+    if hasattr(cls, "YQL_TEST_LIBRARY"):
+        test_lib_path = os.path.join(cls.Env.configs_path, "test_lib.sql")
+        libraries["test"] = test_lib_path
+        with open(test_lib_path, "w") as fp:
+            fp.write(getattr(cls, "YQL_TEST_LIBRARY"))
+            fp.close()
```
Comment (Kontakter) [inline yt/yql/tests/conftest.py:67]: Explicit `close` is unnsecessary

File: yt/yql/tests/test_simple.py [hunk 4]
```diff
@@ -104,6 +108,12 @@ def test_calc(self, query_tracker, yql_agent):
             select 2 as b, 1 as a
         """, [[{"column0": 1}], [{"column0": 2}], [{"a": 1, "b": 2}]])
 
+    @authors("a-romanov")
+    def test_libs(self, query_tracker, yql_agent):
+        self._test_simple_query("""
+            select core::IndexOf([3,7,1], 7) as idx, test::my_sqr(3) as sqr;
+        """, [{'idx': 1, "sqr": 9}])
```
Comment (Kontakter) [inline yt/yql/tests/test_simple.py:115]: Use double quotes please: `'idx'` -> `"idx"`

**Review by Krisha11 — APPROVED** [review]

**Comment by Kontakter** [comment]
New test has failed:
```
yt/yql/tests/test_simple.py:113: in test_libs
    self._test_simple_query("""
yt/yql/tests/test_simple.py:38: in _test_simple_query
    result = self._run_simple_query(query, **kwargs)
yt/yql/tests/test_simple.py:28: in _run_simple_query
    query.track()
yt/yt/tests/library/yt_queries.py:42: in track
    raise YtError.from_dict(query["error"])
 yt.common.YtError: Query ea691d8-c07b2361-529d745d-f4c9d254 failed
E       YQL plugin call failed
E           There are some issues
E               Type annotation
E                   At function: RemovePrefixMembers, At function: Unordered, At function: PersistableRepr, At function: OrderedSqlProject, At function: SqlProjectItem
E                       At function: Apply, At function: Udf, At test.my_sqr
E                           Failed to find UDF function: test.my_sqr, reason: Error: Module test is not registered
E   
E   ***** Details:
E   Query ea691d8-c07b2361-529d745d-f4c9d254 failed    
E       origin          localhost on 2024-10-12T11:29:57.301179Z (pid 16188, tid d7ae619e7f178c58, fid fffeeea2be042ef7)    
E       thread          Control    
E       trace_id        ba635b5-dd5b65a1-44ba40a7-a2757f37    
E       span_id         17175465361540233154    
E       query_id        ea691d8-c07b2361-529d745d-f4c9d254
E   YQL plugin call failed    
E       origin          localhost on 2024-10-12T11:29:56.833302Z (pid 16246, tid 2ebf2fd7b25d3cac, fid fffeeea2e1a7bebe)    
E       thread          Yql:5    
E       trace_id        ba635b5-dd5b65a1-44ba40a7-a2757f37    
E       span_id         10121668854761443109    
E       request_id      92ad192-148b1bca-1c4f4cab-9f352235    
E       connection_id   5e17a6c2-73dba13d-e5b61d9c-ae0d2d8a    
E       verification_mode none    
E       realm_id        0-0-0-0    
E       connection_type client    
E       timeout         86400000    
E       method          StartQuery    
E       address         localhost:19576    
E       encryption_mode optional    
E       service         TYqlService
E   There are some issues    
E       origin          localhost on 2024-10-12T11:29:57.631910Z
E   Type annotation    
E       code            31030    
E       origin          localhost on 2024-10-12T11:29:57.631894Z    
E       yql_status      Core type ann    
E       severity        Error
E   At function: RemovePrefixMembers, At function: Unordered, At function: PersistableRepr, At function: OrderedSqlProject, At function: SqlProjectItem    
E       code            30000    
E       origin          localhost on 2024-10-12T11:29:57.631877Z    
E       yql_status      Default error    
E       end_position    {'column': 13, 'row': 2}    
E       severity        Error    
E       start_position  {'column': 13, 'row': 2}
E   At function: Apply, At function: Udf, At test.my_sqr    
E       code            30000    
E       origin          localhost on 2024-10-12T11:29:57.631844Z    
E       yql_status      Default error    
E       end_position    {'column': 60, 'row': 2}    
E       severity        Error    
E       start_position  {'column': 60, 'row': 2}
E   Failed to find UDF function: test.my_sqr, reason: Error: Module test is not registered    
E       code            30000    
E       origin          localhost on 2024-10-12T11:29:57.631790Z    
E       yql_status      Default error    
E       end_position    {'column': 60, 'row': 2}    
E       severity        Error    
E       start_position  {'column': 60, 'row': 2}
```

**Comment by Tony-Romanov** [comment]
It looks like the old version libyqlplugin.so was used.
How to check it?

**Comment by Tony-Romanov** [comment]
I did it in a separate PR https://github.com/ytsaurus/ytsaurus/pull/880
Or did i something wrong?

**Comment by gritukan** [comment]
@Krisha11, did you update sandbox resource for internal tests run for the aforementioned commit?

**Comment by Krisha11** [comment]
Yes, this failure was because of old sandbox resource. Everything is OK.

**Pull request closed without merging**